### PR TITLE
gdu 2.2.0 (new formula)

### PR DESCRIPTION
**Pull request by dundee**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/gdu.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  homepage "https://github.com/dundee/gdu"
+  url "https://github.com/dundee/gdu/archive/v2.2.0.tar.gz"
+  sha256 "94f4faaee6c1676b73bb75f031cc3d004b8df713fb12d1dfc90ed592bf302df7"
+  license "MIT"
+
+  depends_on "go" => :build
+
+  def install
+    system "go", "build", "-ldflags=\"-s -w -X 'main.AppVersion=v2.2.0'\"", "."
+    bin.install "gdu" => "gdu"
```
Comment (carlocab) [inline Formula/gdu.rb:12]:
```suggestion
    bin.install "gdu"
```

File: Formula/gdu.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  desc "Disk usage analyzer with console interface written in Go"
+  homepage "https://github.com/dundee/gdu"
+  url "https://github.com/dundee/gdu/archive/v2.2.0.tar.gz"
+  sha256 "94f4faaee6c1676b73bb75f031cc3d004b8df713fb12d1dfc90ed592bf302df7"
+  license "MIT"
+
+  depends_on "go" => :build
+
+  def install
+    system "go", "build", "-ldflags=\"-s -w -X 'main.AppVersion=v2.2.0'\"", "."
```
Comment (carlocab) [inline Formula/gdu.rb:11]:
```suggestion
    system "go", "build", *std_go_args, "-ldflags", "-s -w", "-X main.AppVersion=v#{version}"
```
I think this works. If it does you can drop the `bin.install` line.

File: Formula/gdu.rb [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    bin.install "gdu" => "gdu"
+  end
+
+  test do
+    system "#{bin}/gdu", "-v"
```
Comment (carlocab) [inline Formula/gdu.rb:16]: Version tests rarely ever error out when they need to, so we'll need a more substantial test than this. Cf. https://docs.brew.sh/Formula-Cookbook#add-a-test-to-the-formula

File: Formula/gdu.rb (hunk 2 above)
Comment (dundee) [reply Formula/gdu.rb]: Thanks, `std_go_args` works, `bin.install` dropped.

File: Formula/gdu.rb (hunk 3 above)
Comment (dundee) [reply Formula/gdu.rb:16]: Ok, I will add more tests. Thanks!

File: Formula/gdu.rb [hunk 4]
```diff
@@ -0,0 +1,18 @@
+class Gdu < Formula
+  desc "Disk usage analyzer with console interface written in Go"
+  homepage "https://github.com/dundee/gdu"
+  version "2.2.0"
+  url "https://github.com/dundee/gdu/archive/v#{version}.tar.gz"
```
Comment (carlocab) [inline Formula/gdu.rb:5]:
```suggestion
  url "https://github.com/dundee/gdu/archive/v2.2.0.tar.gz"
```
The version is automatically parsed from the URL so there is no need to specify it.

File: Formula/gdu.rb [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+  sha256 "94f4faaee6c1676b73bb75f031cc3d004b8df713fb12d1dfc90ed592bf302df7"
+  license "MIT"
+
+  depends_on "go" => :build
+
+  def install
+    system "go", "build", *std_go_args, "-ldflags=\"-s -w -X 'main.AppVersion=v#{version}'\"", "."
```
Comment (dawidd6) [inline Formula/gdu.rb:12]:
```suggestion
    system "go", "build", *std_go_args, "-ldflags", "-s -w -X main.AppVersion=v#{version}"
```

File: Formula/gdu.rb (hunk 4 above)
Comment (dundee) [reply Formula/gdu.rb:5]: Cool, thanks.

File: Formula/gdu.rb (hunk 5 above)
Comment (dundee) [reply Formula/gdu.rb:12]: This works, thank you!

File: Formula/gdu.rb (hunk 3 above)
Comment (dundee) [reply Formula/gdu.rb:15]: I added some basic test for content of help. All the other features unfortunately require user input.

Comment (carlocab) [reply Formula/gdu.rb:15]: Can you write some text into a file and check the usage of that file or the directory that contains it? https://github.com/dundee/gdu#usage

If you need some other content to populate the test directory, you can use the text fixtures: https://github.com/Homebrew/brew/tree/master/Library/Homebrew/test/support/fixtures

Comment (dawidd6) [reply Formula/gdu.rb:15]: Could you also assert the output of gdu version? So we can be sure there is not breakage ever of version variable setting at build time.

File: Formula/gdu.rb (hunk 5 above)
Comment (dawidd6) [reply Formula/gdu.rb]: The last argument (dot) is unneeded too btw.

File: Formula/gdu.rb (hunk 3 above)
Comment (dundee) [reply Formula/gdu.rb:15]: > Can you write some text into a file and check the usage of that file or the directory that contains it? https://github.com/dundee/gdu#usage
> 
> If you need some other content to populate the test directory, you can use the text fixtures: https://github.com/Homebrew/brew/tree/master/Library/Homebrew/test/support/fixtures

I cannot right now. Gdu has no text output, only GUI.

Comment (dundee) [reply Formula/gdu.rb]: > Could you also assert the output of gdu version? So we can be sure there is no breakage ever of version variable set at build time.

Assert added, thanks!

File: Formula/gdu.rb (hunk 5 above)
Comment (dundee) [reply Formula/gdu.rb]: > The last argument (dot) is unneeded too btw.

It works. Dot removed. Thank you!

File: Formula/gdu.rb (hunk 3 above)
Comment (carlocab) [reply Formula/gdu.rb]: Don't check the output then; we just need to check that it'll run without producing an error.

**Review by dawidd6 — APPROVED** [review]

**Review by chenrui333 — APPROVED** [review]
👍

**Review by chenrui333 — APPROVED** [review]

**Comment by chenrui333** [comment]
@dundee, thanks for your contribution to Homebrew! :tada: 🥇

Without awesome contributors like you, it would be impossible to maintain Homebrew to the high level of quality users have come to expect. Thank you!!!!

**Comment by dundee** [comment]
Guys @chenrui333, @carlocab, @dawidd6, thanks for your help, patience and valuable advice! 

Homebrew is very pleasant community to contribute to.

**Pull request closed without merging**